### DISCUSSION TITLE
contact form vs. email address

### DISCUSSION THREAD

**Discussion by yekibud** [GetPublii/Publii#1853] [Help] [answered]
So I found [this Publii article](https://getpublii.com/blog/does-your-website-need-a-contact-form.html) suggesting to just use an email address in most cases _instead of_ a contact form, but it doesn't address the spam issue of email addresses which, from my experience, was one of the main purposes of using contact forms because you can captcha and other stuff to limit submissions - whereas exposing your email address in plain text was spider/bot bait for increasing spam.  Am I just behind the times on email server spam detection, and is it a safe and sane thing to use plain text email contacts on your website these days?  Side note - why no Publii chatroom (a la Matrix)?

**Comment by midlevels** [comment]
Have you looked at services like FormCarry? (they have a free service for smaller sites)

[FormCarry](https://formcarry.com/form/html-contact-form)

**Reply by yekibud** [reply]
Thanks, but the question is about why Publii is recommending email instead of contact forms.  I understand there are ways to use contact forms with Publii.

**Comment by candidexmedia** [comment]
I'm also iffy about having your email out there because of spam, even though contact forms aren't immune to spam. Gmail and Yahoo! have also tightened up antispam measures earlier this year with DKIM, DMARC, and whatnot.

I think the article does a pretty good job of outlining the pros, cons, and nuance of forms vs emails. The conclusion I got was that: it's up to you. Use the method that suits your needs.

The only thing I would say is that there are many more form processors out there besides the ones listed in the article, and some can even inherit the form styling, or allow you to override the style. I'm partial to FormSubmit (it's free, unlimited submissions, inherits styling), though I've seen folks here use Tally as well. There are even ways to [create a form that uses Google Forms as a backend, but takes on your styling and form fields](https://blog.webjeda.com/google-form-customize/) (rather than an iframe).

>  Side note - why no Publii chatroom (a la Matrix)?

What advantage would a chatroom provide over this forum?

**Reply by yekibud** [reply]
Thanks for your reply @candidexmedia, but I still think there is a gap in the article regarding the issue of spam, which IIUC is still more best practice to use a contact form to prevent as a first line of defense, even though they are not completely immune.  IMO this article seems to be encouraging users to be comfortable with the easy path of just adding their email address in plain text to their Publii site, which I think could lead to a greater quantity of spam, not to mention doxing.  New and less technical users may not have the same reservation as us about publishing their personal email in plain text, so to me this is a concern which should be addressed in the article - unless I'm somehow wrong about my two points (more spam and doxing).

Regarding the advantage of a Publii chatroom - it's the same advantage as any chatroom for a software project with a reasonably sized community: day-to-day discussions, quick support, and real-time collaboration.  It seems to me that blogging software like Publii is more on the social/community side of the software spectrum and could benefit from a chatroom - but I understand not all projects have chatrooms.

**Reply by candidexmedia** [reply, marked as answer]
Hopefully, the dev team can explain their reasoning and address all your points. I'm not sure that they check every single thread (since, iirc, they both work full-time outside of Publii), but they do sometimes pop up.

>  IMO this article seems to be encouraging users to be comfortable with the easy path of just adding their email address in plain text to their Publii site, which I think could lead to a greater quantity of spam, not to mention doxing.

You're right that not everyone is aware of the risks of public emails on websites. That said, the article does mention using a throwaway address (or what they call an email that's dedicated *just* to this), and providing other alternatives (FB Messenger, etc.)

Spam-wise: it would be good to hear from folks who do put their address on their site, and what their experience is like. From a quick search in tech forums, it doesn't seem to be as much of an issue? (still wouldn't personally do it lol)

Chat-wise: the social aspect makes sense, and it would be better for instant chit chat. From my experience, though, I find chat rooms above a certain size a nightmare to navigate, especially to get help. I tend to mute most of the FOSS chatrooms I'm in (except announcements), and prefer forums to give help requests their own space, offer better indexability, and richer reply formats like attaching images, videos, and formatted code blocks.

From the dev/maintainer perspective: it just adds one more platform to monitor, manage, and moderate... I find that Matrix, in particular, doesn't have moderation tools that are as robust as Discord.

That said, I don't see why a community member couldn't create an unofficial chatroom for Publii bloggers. There are already several third-party unnoficial plugins out there.

**Reply by yekibud** [reply]
Oh, gosh - I think I glossed over the point of providing a _dedicated_ email address for the contact email, thanks for pointing that out @candidexmedia, that makes the doxing point moot.  And the article _does_ mention that as sort of a way to control spam - although then you've just got another inbox for spam, but better than your personal inbox.  So, heck, maybe I'll give a dedicated plain-text email a try and see if my mail server blows up.

For the chat room, the only problem with unofficial would be discoverability - unless the devs would be willing to link an unofficial chat room on the project website - although I guess that would make it official?  Or maybe the unofficial label is still useful to distance devs from responsibility.

**Reply by candidexmedia** [reply]
>  So, heck, maybe I'll give a dedicated plain-text email a try and see if my mail server blows up.

If you do, please report back 😅

**Comment by bobmitro** [comment]
Hi guys, long story short, I won’t dive into the debate. But if you want to use emails, I’ve created a plugin that obfuscates email addresses by reversing the `user` and `domain` parts and storing them in `data-user` and `data-domain` attributes. The email is dynamically reconstructed in the browser using JavaScript. This ensures emails aren’t exposed in the raw HTML, offering basic protection against email scraping bots.  

However, this will not protect against bots that parse JavaScript, but it is still better than leaving emails fully exposed.  

The plugin works globally by detecting every email format, plain text, or `mailto` link or targeting only specific email addresses defined by the user in the plugin’s settings. Additionally, if an email appears at the beginning of a post and ends up in the meta description (since the beginning of the post is treated as the excerpt), the plugin completely removes the email from the <head> section.

Feel free to check it out; it might be useful for your projects! 😊

Download: [Email Address Obfuscation](https://marketplace.getpublii.com/plugins/email-address-obfuscation/)

**Reply by candidexmedia** [reply]
Anazing, thank you!!

**Reply by bobmitro** [reply]
Check out this one. It's the same 1.0.0 version (install it again), but this is a much more advanced plugin than before.

Download: [Email Address Obfuscation](https://marketplace.getpublii.com/plugins/email-address-obfuscation/)

**Reply by Verhoeckx** [reply]
Wow, this plug-in is very good!! Thank you very much!!

**Comment by Ainz** [comment]
I still want a form and going with Tally.

E-mal address obfuscation is great, but at the end of the day you want submitted data checked and organized somehow.
